meshmode/array_context.py

```diff
@@ -463,20 +468,28 @@ class DataclassArrayContainerWithArithmetic(
 
 @serialize_container.register(DataclassArrayContainer)
 def _(ary: DataclassArrayContainer):
-    return ((var.name, getattr(ary, var.name)) for var in fields(ary))
+    # FIXME: These field lists could be generated statically.
+    z = [(fld.name, getattr(ary, fld.name))
+            for fld in fields(ary)
+            if issubclass(fld.type, (ArrayContainer, NumpyObjectArray))]
```

Comment:
Is `fld.type` the actual type of the field or the annotation?

Reply:
[Docs](https://docs.python.org/3/library/dataclasses.html#dataclasses.Field) say "type of the field", so I assume it's the actual type (?).

Reply:
If you call `fields(type)`, it appears to be the declared type.